orbstation/antagonists/traitor/objectives/final_objective/malware_injection.dm

```diff
@@ -0,0 +1,417 @@
+			return
+		if(!current)
+			to_chat(user, span_alert("You haven't selected anything to inject malware into!"))
+			return
+		if(!inject_check())
+			to_chat(user, span_alert("Upload failed! Check to make sure [current.name] is functioning properly."))
+			current = null
+			return
+		var/area/comp_area = get_area(src)
+		var/area/req_area = get_req_area(user)
+		if(!req_area) // you don't have the malware injection objective
+			to_chat(user, span_alert("Access denied."))
+			return
+		if(!istype(comp_area, req_area))
+			to_chat(user, span_alert("Upload failed! Unable to establish connection to [current.name] in this area."))
+			return
+		begin_injection(M, user)
+	else
+		return ..()
+
+/obj/machinery/computer/malware/screwdriver_act(mob/living/user, obj/item/I)
+	if(injecting)
+		to_chat(user, span_alert("ERROR: Malware injection in progress."))
+		return FALSE
+	return ..()
+
+/obj/machinery/computer/malware/proc/get_req_area(mob/living/user)
+	if(!user.mind)
+		return
+
+	for(var/datum/traitor_objective/final/malware_injection/malware_objective in SStraitor.taken_objectives_by_type[/datum/traitor_objective/final/malware_injection])
+		var/datum/uplink_handler/handler = malware_objective.handler
+		if(handler.owner != user.mind)
+			continue
+
+		return malware_objective.console_area
+
+	return
+
+/obj/machinery/computer/malware/proc/inject_check()
+	if(!current)
+		return FALSE
+	if(!isAI(current) || IS_MALF_AI(current) || current.stat == DEAD)
+		return FALSE
+	if(!circuit || machine_stat & (NOPOWER|BROKEN|MAINT))
+		return FALSE
+	return TRUE
+
+/obj/machinery/computer/malware/proc/begin_injection(obj/item/malware_injector/M, mob/living/user)
+	if(M.hacking || injecting) // just in case
+		return
+
+	user.visible_message(span_danger("[user] holds [M] up to [src] and begins pressing buttons!"), span_notice("You activate [M] and begin transferring the virus to [src]. Keep [M] in your active hand and stand still."))
+	M.hacking = TRUE
+	injecting = TRUE
+
+	if(!do_after(user, INJECTOR_HACK_TIME, src, IGNORE_SLOWDOWNS, extra_checks = CALLBACK(src, .proc/inject_check)))
+		playsound(user, 'sound/machines/buzz-sigh.ogg', 30, TRUE)
+		to_chat(user, span_warning("Transfer aborted."))
+		M.hacking = FALSE
+		injecting = FALSE
+		return
+
+	M.hacking = FALSE
+	M.spent = TRUE
+	hacked_apcs = M.hacked_apcs.Copy()
+	to_chat(user, span_boldnotice("Transfer complete. [src] begins injecting the virus into [current]'s mainframe!"))
+	// time to go loud!
+	trigger_alarm(user) // set off the alarm, close and lock all doors in the area, alert the AI in big bold letters
+	announce_corruption()
+	advance_corruption_stage(user)
+
+/obj/machinery/computer/malware/proc/advance_corruption_stage(mob/living/user)
+	if(!do_after(user, TIME_PER_STAGE, src, IGNORE_USER_LOC_CHANGE | IGNORE_TARGET_LOC_CHANGE | IGNORE_HELD_ITEM | IGNORE_INCAPACITATED | IGNORE_SLOWDOWNS, extra_checks = CALLBACK(src, .proc/inject_check)))
+		if(!machine_stat & (BROKEN|NOPOWER|MAINT))
```

Comment:
I think the linter wants these to be `&&`

Reply:
I'm pretty sure this is a bitwise operation though, it's checking to see if machine_stat has any of those flags. I think it wants me to put parentheses around it and put the exclamation mark outside of the parentheses